hedera-node/src/main/java/com/hedera/services/queries/token/GetTokenNftInfosAnswer.java

```diff
@@ -0,0 +1,187 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import com.hedera.services.context.primitives.StateView;
+import com.hedera.services.queries.AnswerService;
+import com.hedera.services.state.enums.TokenType;
+import com.hedera.services.txns.validation.OptionValidator;
+import com.hedera.services.utils.SignedTxnAccessor;
+import com.hederahashgraph.api.proto.java.HederaFunctionality;
+import com.hederahashgraph.api.proto.java.Query;
+import com.hederahashgraph.api.proto.java.Response;
+import com.hederahashgraph.api.proto.java.ResponseCodeEnum;
+import com.hederahashgraph.api.proto.java.TokenGetNftInfosQuery;
+import com.hederahashgraph.api.proto.java.TokenGetNftInfosResponse;
+import com.hederahashgraph.api.proto.java.TokenNftInfo;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static com.hedera.services.utils.SignedTxnAccessor.uncheckedFrom;
+import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.INVALID_NFT_ID;
+import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.INVALID_QUERY_RANGE;
+import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.INVALID_TOKEN_ID;
+import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.NOT_SUPPORTED;
+import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.OK;
+import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.TOKEN_WAS_DELETED;
+import static com.hederahashgraph.api.proto.java.ResponseType.COST_ANSWER;
+
+public class GetTokenNftInfosAnswer implements AnswerService {
+
+	public static final String TOKEN_NFT_INFOS_CTX_KEY = GetTokenNftInfosAnswer.class.getSimpleName() + "_tokenNftInfos";
+
+	private OptionValidator validator;
+
+	public GetTokenNftInfosAnswer(OptionValidator validator) {
+		this.validator = validator;
+	}
+
+	@Override
+	public boolean needsAnswerOnlyCost(Query query) {
+		return COST_ANSWER == query.getTokenGetNftInfos().getHeader().getResponseType();
+	}
+
+	@Override
+	public boolean requiresNodePayment(Query query) {
+		return typicallyRequiresNodePayment(query.getTokenGetNftInfos().getHeader().getResponseType());
+	}
+
+	@Override
+	public Response responseGiven(Query query, StateView view, ResponseCodeEnum validity, long cost) {
+		return responseFor(query, view, validity, cost, NO_QUERY_CTX);
+	}
+
+	@Override
+	public Response responseGiven(
+			Query query, StateView view, ResponseCodeEnum validity, long cost, Map<String, Object> queryCtx) {
+		return responseFor(query, view, validity, cost, Optional.of(queryCtx));
+	}
+
+	@Override
+	public ResponseCodeEnum checkValidity(Query query, StateView view) {
+		var tokenNftInfosQuery = query.getTokenGetNftInfos();
+		var id = tokenNftInfosQuery.getTokenID();
+
+		var optionalInfo = view.infoForToken(id);
```

Comment:
`infoForToken` is pretty big operation, populating a lot of stuff. I think that it would be better to have a lightweight check. There is `tokenExists` method, but unfortunately, it only checks for existence. Ideally we would like to check for the existence but include the check for deleted as-well.

I think that it would be best to use the `tokenWith` method and operate on that